### PR TITLE
Set JDK to 11

### LINKED ISSUE #226: [4.0] Build fails with a JDK 8
Although the build claims to require [only a JDK 8](https://github.com/neo4j-contrib/neo4j-jdbc/blob/03886f20976c6ab1f2379dab6cec2d6b4ea96fcf/pom.xml#L78), it fails with such:

```shell
$ mvn clean package -DskipTests
[...]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project neo4j-jdbc-bolt: Compilation failure
[ERROR] /Users/fbiville/workspace/neo4j-jdbc/neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jSetParameterDateIT.java:[7,25] cannot access org.neo4j.graphdb.Result
[ERROR]   bad class file: /Users/fbiville/.m2/repository/org/neo4j/neo4j-graphdb-api/4.0.0/neo4j-graphdb-api-4.0.0.jar(org/neo4j/graphdb/Result.class)
[ERROR]     class file has wrong version 55.0, should be 52.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
```

No issue with a JDK 11.

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #226

### CODE REVIEW

File: neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jDriverTest.java [hunk 1]
```diff
@@ -217,13 +217,13 @@
...
-			assertEquals(true, config.encrypted());
-			assertEquals(false, config.logLeakedSessions());
+			assertEquals((int) TimeUnit.SECONDS.toMillis( 30 ), config.connectionTimeoutMillis());
+			assertFalse(config.encrypted());
+			assertFalse(config.logLeakedSessions());
 			// assertEquals(Config.LoadBalancingStrategy.LEAST_CONNECTED, config.loadBalancingStrategy());
 			assertEquals(PoolSettings.DEFAULT_MAX_CONNECTION_LIFETIME, config.maxConnectionLifetimeMillis());
 			assertEquals(PoolSettings.DEFAULT_MAX_CONNECTION_POOL_SIZE, config.maxConnectionPoolSize());
-			assertEquals(Config.TrustStrategy.trustAllCertificates().strategy(), config.trustStrategy().strategy());
+			assertEquals(Config.TrustStrategy.trustSystemCertificates().strategy(), config.trustStrategy().strategy());
```
Comment (conker84) [inline neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jDriverTest.java]: Why this changed?

**Review by conker84 — COMMENTED** [review]
just a comment otherwise looks good to me!

File: neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jDriverTest.java (hunk 1 above)
Comment (fbiville) [reply neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jDriverTest.java]: This is a 4.0 change, as documented [here](https://github.com/neo4j/neo4j-java-driver/wiki/4.0-changelog#400-beta01). The change is [here](https://github.com/neo4j/neo4j-java-driver/pull/620).

Comment (fbiville) [reply neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/BoltNeo4jDriverTest.java]: After 1 hour of looking in the driver code and the changelogs, I do not understand how the build was green before.
I started a new 4.0 build on Travis, but it's taking ages to provision a VM: https://travis-ci.org/github/neo4j-contrib/neo4j-jdbc/builds/743243669.

**Comment by conker84** [comment]
@fbiville thank you so much

**Pull request merged**